Cargo.toml

```diff
@@ -63,6 +63,10 @@ near-chain = { path = "./chain/chain" }
 
 node-runtime = { path = "./runtime/runtime" }
 
+near-vm-logic = { path = "./runtime/near-vm-logic" }
+near-vm-errors = { path = "./runtime/near-vm-errors" }
+near-vm-runner = { path = "./runtime/near-vm-runner" }
```

Comment:
Hm, wait, why we need these here?

Reply:
Ah, right, thats' because we add a test to the root package. Naively, I'd expect the test to be within `near-vm-runner` crate, like this:

```diff
diff --git a/Cargo.lock b/Cargo.lock
index 032d7eb9..60f5dd02 100644
Binary files a/Cargo.lock and b/Cargo.lock differ
diff --git a/Cargo.toml b/Cargo.toml
index b19e030e..1dd38eb4 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -63,10 +63,6 @@ near-chain = { path = "./chain/chain" }
 
 node-runtime = { path = "./runtime/runtime" }
 
-near-vm-logic = { path = "./runtime/near-vm-logic" }
-near-vm-errors = { path = "./runtime/near-vm-errors" }
-near-vm-runner = { path = "./runtime/near-vm-runner" }
-
 near-jsonrpc = { path = "./chain/jsonrpc" }
 near-network = { path = "./chain/network" }
 
diff --git a/runtime/near-vm-runner/src/tests.rs b/runtime/near-vm-runner/src/tests.rs
index 34d6636a..7cbaf3cf 100644
--- a/runtime/near-vm-runner/src/tests.rs
+++ b/runtime/near-vm-runner/src/tests.rs
@@ -1,5 +1,6 @@
 mod error_cases;
 mod invalid_contracts;
+mod contract_preload;
 mod rs_contract;
 mod ts_contract;
 
diff --git a/tests/test_vm_runner.rs b/runtime/near-vm-runner/src/tests/contract_preload.rs
similarity index 96%
rename from tests/test_vm_runner.rs
rename to runtime/near-vm-runner/src/tests/contract_preload.rs
index 1cd42463..3ece2215 100644
--- a/tests/test_vm_runner.rs
+++ b/runtime/near-vm-runner/src/tests/contract_preload.rs
@@ -1,7 +1,6 @@
 use near_primitives::hash::hash;
 use near_primitives::runtime::fees::RuntimeFeesConfig;
 use near_vm_logic::{ProtocolVersion, VMConfig, VMContext, VMKind, VMOutcome};
-use near_vm_runner::{run_vm, ContractCallPrepareRequest, ContractCaller, VMError};
 
 use near_primitives::borsh::BorshSerialize;
 use near_primitives::types::CompiledContractCache;
@@ -13,10 +12,13 @@ use std::sync::{Arc, Mutex};
 use std::thread::sleep;
 use std::time::Duration;
 
+use crate::{run_vm, ContractCallPrepareRequest, ContractCaller, VMError};
+
+
 const TEST_CONTRACT_1: &'static [u8] =
-    include_bytes!("../runtime/near-vm-runner/tests/res/test_contract_rs.wasm");
+    include_bytes!("../../tests/res/test_contract_rs.wasm");
 const TEST_CONTRACT_2: &'static [u8] =
-    include_bytes!("../runtime/near-vm-runner/tests/res/test_contract_ts.wasm");
+    include_bytes!("../../tests/res/test_contract_ts.wasm");
 
 fn default_vm_context() -> VMContext {
     return VMContext {
```

But I don't actually know the rule we use to decide if we want to put a test into the top-level tests directory, or into a specific crate.

Reply:
Actually that's exactly how it used to be, and you suggested to move it here in earlier comments. I'm kind of OK with either approach, but keep moving it forward and backward doesn't make much sense.

Reply:
OK, done